File: online-judge-solutions/Codeforces/1515B.cpp
//https://codeforces.com/problemset/problem/1515/B
//Lang		:	GNU C++14
//Time		:	62 ms
//Memory	:	0 KB
#include<iostream>
#include<math.h>
using namespace std;
bool EsCuad(int x){
  int y=sqrt(x);
  return y*y==x;
}
int main(){
	int t,n;
	cin>>t;
	while(t--){
		cin>>n;
		int n2=n/2;
		if((EsCuad(n) || EsCuad(n/2)) && n%2==0) cout<<"YES\n";
		else cout<<"NO\n";
	}
	return 0;
}

File: online-judge-solutions/Codeforces/1515B.py
#https://codeforces.com/problemset/problem/1515/B
#Lang		:	Python 3.8
#Time		:	124 ms
#Memory         :	0 KB
import math
def EsCuad(x: int)->bool:
    if math.sqrt(x).is_integer(): return True
    else: return False
t=int(input())
for x in range(t):
    n=int(input())
    n2=int(n/2)
    if((EsCuad(n) or EsCuad(n/2)) and n%2==0):print("YES")
    else: print("NO")
